src/arithmetic/map_funcs/map_funcs.c

```diff
@@ -57,6 +58,29 @@ SIValue AR_KEYS(SIValue *argv, int argc, void *private_data) {
 	return SI_NullVal();
 }
 
+SIValue AR_PROPERTIES(SIValue *argv, int argc, void *private_data) {
+	ASSERT(argc == 1);
+
+	if(SI_TYPE(argv[0]) == T_NULL) {
+		return SI_NullVal();
+	} else if(SI_TYPE(argv[0]) == T_MAP) {
+		return argv[0];
+	} else if(SI_TYPE(argv[0]) & (T_MAP | SI_GRAPHENTITY)) {
```

Comment:
is T_MAP is redundant?

Reply:
Fixed in [2e22b7b](https://github.com/RedisGraph/RedisGraph/pull/2632/commits/2e22b7b92699fc7027530007f0047022fdca9018)

Reply:
Fixed in [0cc2dbb](https://github.com/RedisGraph/RedisGraph/pull/2632/commits/0cc2dbb4e25954c4ea584931f74c2f50b0c11f28). The if/else was replaced by switch/case